client/lib/plans/features-list.js

```diff
@@ -231,7 +231,7 @@ export const FEATURES_LIST = {
 
 	[ constants.FEATURE_BACKUP_STORAGE_SPACE_UNLIMITED_SIGNUP ]: {
 		getSlug: () => constants.FEATURE_BACKUP_STORAGE_SPACE_UNLIMITED_SIGNUP,
-		getTitle: () => i18n.translate( 'Unlimited Backup Space' ),
+		getTitle: () => i18n.translate( 'Unlimited backup space' ),
```

Comment:
:ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.